sources/packages/web/src/components/common/restriction/AddRestriction.vue

```diff
@@ -51,74 +77,85 @@ export default {
+    const { showDialog, showModal, resolvePromise } = useModalDialog<
+      AssignRestrictionAPIInDTO | false
+    >();
+    const addRestrictionForm = ref({} as VForm);
+    const formModel = reactive({} as AssignRestrictionAPIInDTO);
 
-    const dialogClosed = () => {
-      showDialog.value = false;
-    };
-    const formLoaded = async (form: any) => {
-      formData.value = form;
-      const dropdown = formioUtils.getComponent(form, CATEGORY_KEY);
+    const categoryItems = async () => {
       const categories =
         await RestrictionService.shared.getRestrictionCategories();
-      const options = [{}];
       // Restriction category Designation is exclusively for Institution. Rest of them are for Student.
       if (props.entityType === RestrictionEntityType.Student) {
         for (const category of categories) {
-          options.push({
-            label: category.description,
+          items.value.push({
+            title: category.description,
             value: category.description,
           });
         }
       } else {
-        options.push({
-          label: "Designation",
+        items.value.push({
+          title: "Designation",
           value: "Designation",
         });
       }
+    };
+
+    onMounted(async () => {
+      categoryItems();
+    });
 
-      dropdown.component.data.values = options;
-      dropdown.redraw();
+    const categoryChanged = (category: string) => {
+      categoryReasonItems(category);
     };
 
-    const formChanged = async (form: any, event: any) => {
-      if (event.changed?.component.key === CATEGORY_KEY) {
-        const selectedRestrictionCategory: string =
-          formioUtils.getComponentValueByKey(form, CATEGORY_KEY);
-        formioDataLoader.loadRestrictionReasons(
-          form,
-          "restrictionId",
-          selectedRestrictionCategory,
-        );
+    const categoryReasonItems = async (category: string) => {
+      reasonItems.value = [];
+      const reasons = await RestrictionService.shared.getRestrictionReasons(
+        category,
+      );
+      // Restriction category Designation is exclusively for Institution. Rest of them are for Student.
+      for (const reason of reasons) {
+        reasonItems.value.push({
+          title: reason.description,
+          value: reason.id,
+        });
       }
     };
-    const submitForm = async () => {
-      return formData.value.submit();
-    };
-    const submitRestriction = async (data: AssignRestrictionAPIInDTO) => {
-      context.emit("submitRestrictionData", data);
-    };
-    const addRestriction = async () => {
-      const formSubmitted = await submitForm();
-      if (formSubmitted) {
-        showDialog.value = false;
+
+    const submit = async () => {
+      const validationResult = await addRestrictionForm.value.validate();
+      if (!validationResult.valid) {
+        return;
       }
+      const payload = { ...formModel };
+      resolvePromise(payload);
```

Comment:
resolvePromise(formModel); ?

Reply:
Resest is not happening, so passing it to a value and resolving it

Reply:
I got it after our chat. Please disregard my last comment.